packages/m365_defender/data_stream/alert/elasticsearch/ingest_pipeline/default.yml

```
@@ -0,0 +1,2318 @@
+---
+description: Pipeline for processing Alert logs.
+processors:
+  - set:
+      field: ecs.version
+      value: '8.4.0'
+  - rename:
+      field: message
+      target_field: event.original
+      ignore_missing: true
+  - json:
+      field: event.original
+      target_field: json
+  - fingerprint:
+      fields:
+        - json.id
+        - json.lastUpdateDateTime
+        - json.incidentId
+        - json.providerAlertId
+      target_field: _id
+      ignore_missing: true
+  - set:
+      field: event.kind
+      value: alert
+  - script:
+      description: Dynamically map event.category and event.type field from evidence.
+      lang: painless
+      if: ctx.json?.evidence instanceof List
+      ignore_failure: true
+      source:
+        def eventCategory = new HashSet();
+        def eventType = new HashSet();
+        for (evidence in ctx.json.evidence) {
+          if (['#microsoft.graph.security.analyzedMessageEvidence', '#microsoft.graph.security.mailboxEvidence', '#microsoft.graph.security.mailClusterEvidence'].contains(evidence["@odata.type"])) {
+            eventCategory.add('email');
+            eventType.add('info');
+          } else if (['#microsoft.graph.security.deviceEvidence'].contains(evidence["@odata.type"])) {
```

Query the used of `ArrayList.contains` when the array has only a single element. Is there a reason not to use straight equality?
```suggestion
          } else if (evidence["@odata.type"] == '#microsoft.graph.security.deviceEvidence') {
```
Similar below.